src/plotting.jl

```diff
@@ -30,81 +30,95 @@ end
     end
     if show_data
-        plot_data!(gp.x, gp.y; ms)
+        inds = [i for (i, d) in enumerate(inds) if d == :]
+        plot_data!(gp.x, gp.y; ms, inds)
     end
     if tight
         return plot!(cbar=false, ticks=false, xlabel="", ylabel="", title="", size=(400,400))
     else
-        return plot!(xlabel="$(models[1].name)", ylabel="$(models[2].name)", title="hard failure boundary", size=(450,400))
+        return plot!(xlabel="$(full_models[1].name)", ylabel="$(full_models[2].name)", title="hard failure boundary", size=(450,400))
     end
 end
 
 
 """
 Plot true function `f`
 """
-function plot_truth(sparams, models; m=[200,200], overlay=false, overlay_levels=50, use_heatmap=false, tight=false, hard=false, lw=1)
-    f = (x1,x2; fx=System.evaluate(sparams, [[x1,x2]])[1]) -> hard ? fx >= 0.5 : fx
+function plot_truth(sparams, models; inds=[:, :, fill(1, length(models) - 2)...], num_steps=200, m=fill(num_steps, length(models)), overlay=false, overlay_levels=50, use_heatmap=false, tight=false, hard=false, lw=1)
     model_ranges = get_model_ranges(models, m)
+    s1, s2 = findall(isequal(:), inds)
+    function f(x...)
+        fx = @suppress System.evaluate(sparams, [x])[1]
+        return hard ? fx >= 0.5 : fx
+    end
+    params = [collect(r[d]) for (r, d) in zip(model_ranges, inds)]
+    params[s1] = params[s1]'
+    y = f.(params...)
+
     plot_f = use_heatmap ? heatmap : contourf
-    @suppress plot_f(model_ranges[1], model_ranges[2], f, c=COLOR_FAIL, lc=:white, clims=(0,1), lw=lw)
+    plot_f(model_ranges[s1], model_ranges[s2], y, c=COLOR_FAIL, lc=:white, clims=(0,1), lw=lw)
     if overlay
-        p(x,y) = pdf(models[1].distribution, x) * pdf(models[2].distribution, y)
-        @suppress contour!(model_ranges[1], model_ranges[2], p, c=cgrad([:gray, :black, :black, :white], 10, categorical=true, scale=:exp, rev=false), levels=overlay_levels)
+        p(x,y) = pdf(models[s1].distribution, x) * pdf(models[s2].distribution, y)
+        @suppress contour!(model_ranges[s1], model_ranges[s2], p, c=cgrad([:gray, :black, :black, :white], 10, categorical=true, scale=:exp, rev=false), levels=overlay_levels)
     end
     if tight
         return plot!(cbar=false, ticks=false, xlabel="", ylabel="", title="", size=(400,400))
     else
-        return plot!(xlabel="\$$(models[1].name)\$", ylabel="\$$(models[2].name)\$", title="truth", size=(450,400))
+        return plot!(xlabel="\$$(models[s1].name)\$", ylabel="\$$(models[s2].name)\$", title="truth", size=(450,400))
     end
 end
 
 
 """
 Plot true function, include data (when passing in `gp`).
 """
-function plot_truth(gp, sparams, models; m=[200,200], ms=4)
-    plot_truth(sparams, models; m)
-    plot_data!(gp.x, gp.y; ms)
+function plot_truth(gp, sparams, models; inds=[:, :, fill(1, length(models) - 2)...], num_steps=200, m=fill(num_steps, length(models)), ms=4)
```

Comment:
Looks like `inds` here does not have the same meaning/dimensions as the one for the other plot function, should we harmonize the interface here?

Reply:
renamed to `plot_inds`